### PR TITLE
Feat/adam/match task

### PR DESCRIPTION
**Pull request by aj-goodman**
• Add simple rake task to recalculate match scores once per day
• Limit matches to 10 non-rejected
• Add Supercronic to Docker
• Update fly.toml deployment to include cron machine
• Add Profile seeds
• Add games to matching profiles.

### CODE REVIEW

File: make_matcher_backend/app/controllers/api/matches_controller.rb [hunk 1]
```diff
@@ -6,8 +6,11 @@ def index
...
 
   def destroy
-    Match.where(matcher_id: current_user.profile.id, matched_id: params[:id]).or(Match.where(matcher_id: params[:id], matched_id: current_user.profile.id)).destroy_all
+    matches =
+      Match.where(matcher_id: current_user.profile.id, matched_id: params[:id])
+         .or(Match.where(matcher_id: params[:id], matched_id: current_user.profile.id))
+    matches&.update_all(reject: true)
 
-    render json: { message: "Success" }, status: :ok
+    render json: { message: matches ? "Success" : "Not found" }, status: matches ? :ok : :not_found
```
Comment (eddof13) [inline make_matcher_backend/app/controllers/api/matches_controller.rb:14]: I don't think that's necessary- delete can be idempotent, does nothing if nothing found

File: make_matcher_backend/app/models/match.rb [hunk 2]
```diff
@@ -7,13 +7,17 @@ class Match < ApplicationRecord
...
-  SCORE_ATTRS = %i[location utc_offset age intensity skill language days times].freeze
+  SCORE_ATTRS = %i[location utc_offset age intensity skill language days times game_ids].freeze
 
   # Calculate Match Score - Lower is better.
   def calculate_score
     self.score = SCORE_ATTRS.map { |attr| score_for(attr) }.sum
   end
 
+  def games
+    matched.games & matcher.games
```
Comment (eddof13) [inline make_matcher_backend/app/models/match.rb:18]: bitwise and?

File: make_matcher_backend/app/models/profile.rb [hunk 3]
```diff
@@ -19,6 +19,9 @@ class Profile < ApplicationRecord
   belongs_to :user
   has_and_belongs_to_many :games
   has_many :matches, foreign_key: :matcher_id
+  def matches
+    super.where(reject: false).order(:score).limit(10)
```
Comment (eddof13) [inline make_matcher_backend/app/models/profile.rb:23]: you can put a scope on the `Match` model- like `matches.not_rejected` as well, maybe clenaer

File: make_matcher_backend/app/controllers/api/matches_controller.rb (hunk 1 above)
Comment (aj-goodman) [reply make_matcher_backend/app/controllers/api/matches_controller.rb:14]: forgot to remove after adding for postman testing... will update

File: make_matcher_backend/app/models/match.rb (hunk 2 above)
Comment (aj-goodman) [reply make_matcher_backend/app/models/match.rb]: `Game.joins(:profiles).where(profiles: { id: [matcher_id, matched_id] })`

File: make_matcher_backend/app/models/profile.rb (hunk 3 above)
Comment (aj-goodman) [reply make_matcher_backend/app/models/profile.rb]: added the scope to the user model to remove matches that are already the user's friend.

`  has_many :matches, through: :profile do
    def daily
      where.not(reject: true, matched_id: proxy_association.owner.friend_ids)
           .order(:score).limit(10)
    end
  end`

**Review by eddof13 — APPROVED** [review]

**Pull request merged**